src/endpoints/sources.js

```diff
@@ -400,100 +400,115 @@ async function sendAny(req, res, url, accept) {
  */
 function receiveSubresource(subresourceExtractor) {
   return async (req, res) => {
-    const method = "PUT";
     const subresource = subresourceExtractor(req);
 
     authz.assertAuthorized(req.user, authz.actions.Write, subresource.resource);
 
-    /* Proxy the data through us:
-     *
-     *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
-     */
-    // eslint-disable-next-line prefer-const
-    let headers = {
-      "Content-Type": subresource.mediaType,
-      ...copyHeaders(req, ["Content-Encoding", "Content-Length"]),
-
-      /* XXX TODO: Consider setting Cache-Control rather than relying on
-       * ambiguous defaults.  Potentially impacts:
-       *
-       *   - Our own fetch() caching, including in sendSubresource() above
-       *   - Our Charon endpoints, if upstream headers are sent to the browser?
-       *   - CloudFront caching (not sure about its default behaviour)
-       *   - Browsers, if fetched directly, such as by redirection
-       *
-       * I think a cautious initial value would be to set "private" or "public"
-       * depending on the Source and then always set "must-revalidate,
-       * proxy-revalidate, max-age=0".  This would allow caches (ours,
-       * browsers, CloudFront?) to store the data but always check upstream
-       * with conditional requests.
-       *   -trs, 7 Dec 2021
-       */
-    };
+    const urlCallback = async (method, headers) => await subresource.url(method, headers);
 
-    // Body of the request as a Node stream
-    let body = req;
+    return receiveAny(req, res,
+      urlCallback,
+      subresource.mediaType
+    );
+  };
+}
 
-    // Compress on the fly to gzip if it's not already gzip compressed.
-    if (!headers["Content-Encoding"]) {
-      delete headers["Content-Length"]; // won't be valid after compression
-      headers["Content-Encoding"] = "gzip";
-      body = body.pipe(zlib.createGzip());
-    }
 
-    if (headers["Content-Encoding"] !== "gzip") {
-      throw new UnsupportedMediaType("unsupported Content-Encoding; only gzip is supported");
-    }
+/**
+ * Proxy the data through us:
+ *
+ *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
+ *
+ * @param {express.request} req - Express-style request instance
+ * @param {express.response} res - Express-style response instance
+ * @param {urlCallback} urlCallback - Callback to retrieve resource URL
+ * @param {string} contentType - `Content-Type` header value to send
+ * @returns {expressEndpointAsync}
+ */
+async function receiveAny(req, res, urlCallback, contentType) {
```

Comment:
Similar to my reservations about `sendAny`'s name, I think this isn't as descriptive a name as it could be. Perhaps `receiveUrl` or `receiveToUpstream` or `proxyToUpstream`?

Reply:
Renamed to `proxyToUpstream` in 50c4c1eeaf8ec8dd6c605a55624165dd48293e42 + 054f05b21089def72659626d866410f40642b706.